### PR TITLE
`Settings` - reveal mnemonic words require pin or LocalAuthentication

### LINKED ISSUE #301: `settings` - download a copy of mnemonic seed
#### What would you like to be added:
Once user has created a wallet. He should be able to download a copy on his mnemonic seed. Wallet Encryption #57 is a dependency of ticket as there should be a security verification before revealing the mnemonic seed.

### PR DESCRIPTION
**Pull request by ivan-zynesis**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Mnemonic words encryption. Reusable mechanic (redux store action) for any action require authentication.

#### Which issue(s) does this PR fixes?:

Fixes #301  

#### Additional comments?:
include #367 and #410

### CODE REVIEW

File: app/components/Button.tsx [hunk 1]
```diff
@@ -8,6 +8,8 @@ interface ButtonProps extends React.PropsWithChildren<TouchableOpacityProps> {
   fill?: 'fill' | 'outline' | 'flat'
   label?: string
   margin?: string
+  delayLongPress?: number
```
Comment (thedoublejay) [inline app/components/Button.tsx:11]: this one is not needed anymore, we have it extended from touchableopacity

**Review by thedoublejay — CHANGES_REQUESTED** [review]

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: app/components/icons/FaceIdIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import * as React from 'react'
+import Svg, { G, Path, SvgProps } from 'react-native-svg'
+
+export function FaceIdIcon (props: SvgProps): JSX.Element {
+  return (
+    <Svg width='64' height='64' viewBox='0 0 64 64' {...props}>
+      <G id='Onboarding' stroke='none' stroke-width='1' fill='none' fill-rule='evenodd'>
```
[the remark this answers is not in the document]
Comment (ivan-zynesis) [reply app/components/icons/FaceIdIcon.tsx]: oops, it seems render fine.

**Comment by thedoublejay** [comment]
@ivan-zynesis on this branch, when creating a wallet from onboarding and going to settings page -> reveal words. The pincode doesn't work.

File: app/components/Button.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ interface ButtonProps extends React.PropsWithChildren<TouchableOpacityProps> {
   fill?: 'fill' | 'outline' | 'flat'
   label?: string
   margin?: string
+  onLongPress?: () => void
```
Comment (thedoublejay) [inline app/components/Button.tsx:11]: this one also is not needed

File: app/components/Button.tsx [hunk 4]
```diff
@@ -25,6 +26,11 @@ export function Button (props: ButtonProps): JSX.Element {
   return (
     <TouchableOpacity
       {...props}
+      onLongPress={e => {
```
Comment (thedoublejay) [inline app/components/Button.tsx:29]: also this, it's already on the TouchableOpacityProps

**Review by thedoublejay — CHANGES_REQUESTED** [review]
For linked PRs, can we point it first to the main branch? It's hard to review with duplicated changes. Hard to see which are the changes for this PR and which is from the other branch.

**Comment by ivan-zynesis** [comment]
> @ivan-zynesis on this branch, when creating a wallet from onboarding and going to settings page -> reveal words. The pincode doesn't work.

Testing/troubleshooting for this.

**Comment by ivan-zynesis** [comment]
> @ivan-zynesis on this branch, when creating a wallet from onboarding and going to settings page -> reveal words. The pincode doesn't work.

Fixed, the encryption only implemented in EnrollBiometric page, bad merge did not navigate there but setWallet directly.

File: app/screens/WalletNavigator/screens/CreateWallet/PinConfirmation.tsx [hunk 5]
```diff
@@ -42,7 +45,11 @@ export function PinConfirmation ({ route }: Props): JSX.Element {
     setTimeout(() => {
       MnemonicEncrypted.toData(copy.words, copy.network, copy.pin)
         .then(async encrypted => {
+          await MnemonicWords.encrypt(words, pin)
```
Comment (ivan-zynesis) [inline app/screens/WalletNavigator/screens/CreateWallet/PinConfirmation.tsx:48]: main change needed here, store it so can be decrypted and revealed later.

File: app/screens/WalletNavigator/screens/CreateWallet/PinConfirmation.tsx [hunk 6]
```diff
@@ -42,7 +45,11 @@ export function PinConfirmation ({ route }: Props): JSX.Element {
...
           await setWallet(encrypted)
+
+          // temp disabled biometric, forward data to biometric enrolment page after more test
+          // navigation.navigate('EnrollBiometric', { pin, encrypted, words })
```
Comment (ivan-zynesis) [inline app/screens/WalletNavigator/screens/CreateWallet/PinConfirmation.tsx:52]: It will remain unused.
No changes required on TransactionAuthorization, it behave as it is when no biometric enrolled.

I think we are confident this code works find and merge-able, but just need more thorough testing for edge cases and different devices.

**Review by thedoublejay — APPROVED** [review]
